pki/src/test/java/org/hyperledger/besu/pki/cms/CmsCreationAndValidationTest.java

```diff
@@ -0,0 +1,183 @@
+  private static final String KEYSTORE_TYPE = "PKCS12";
+  private static final String KEYSTORE_PASSWORD = "validator";
+
+  private KeyStoreWrapper keystore;
+  private KeyStoreWrapper truststore;
+  private CmsValidator cmsValidator;
+
+  @Before
+  public void before() {
+    truststore = loadKeystore("truststore");
+    keystore = loadKeystore("keystore");
+    cmsValidator = new CmsValidator(truststore, loadCRLs("crl.pem"));
+  }
+
+  @Test
+  public void cmsValidationWithTrustedSelfSignedCertificate() {
+    final CmsCreator cmsCreator = new CmsCreator(keystore, "trusted_selfsigned");
+    final Bytes data = Bytes.random(32);
+
+    final Bytes cms = cmsCreator.create(data);
+
+    assertThat(cmsValidator.validate(cms, data)).isTrue();
+  }
+
+  @Test
+  public void cmsValidationWithUntrustedSelfSignedCertificate() {
+    final CmsCreator cmsCreator = new CmsCreator(keystore, "untrusted_selfsigned");
+    final Bytes data = Bytes.random(32);
+
+    final Bytes cms = cmsCreator.create(data);
+
+    assertThat(cmsValidator.validate(cms, data)).isFalse();
+  }
+
+  @Test
+  public void cmsValidationWithValidChain() {
+    final CmsCreator cmsCreator = new CmsCreator(keystore, "trusted");
+    final Bytes data = Bytes.random(32);
+
+    final Bytes cms = cmsCreator.create(data);
+
+    assertThat(cmsValidator.validate(cms, data)).isTrue();
+  }
+
+  @Test
+  public void cmsValidationWithInvalidChain() {
+    final CmsCreator cmsCreator = new CmsCreator(keystore, "untrusted");
+    final Bytes data = Bytes.random(32);
+
+    final Bytes cms = cmsCreator.create(data);
+
+    assertThat(cmsValidator.validate(cms, data)).isFalse();
+  }
+
+  @Test
+  public void cmsValidationWithExpiredCertificate() {
+    final CmsCreator cmsCreator = new CmsCreator(keystore, "expired");
+    final Bytes data = Bytes.random(32);
+
+    final Bytes cms = cmsCreator.create(data);
+
+    assertThat(cmsValidator.validate(cms, data)).isFalse();
+  }
+
+  @Test
+  public void cmsValidationWithRevokedCertificate() {
+    final CmsCreator cmsCreator = new CmsCreator(keystore, "revoked");
+    final Bytes data = Bytes.random(32);
+
+    final Bytes cms = cmsCreator.create(data);
+
+    assertThat(cmsValidator.validate(cms, data)).isFalse();
+  }
+
+  @Test
+  public void cmsValidationWithoutCRLConfigDisablesCRLCheck() {
+    final CmsCreator cmsCreator = new CmsCreator(keystore, "revoked");
+    final Bytes data = Bytes.random(32);
+
+    final Bytes cms = cmsCreator.create(data);
+
+    // Overriding validator with instance without CRL CertStore
+    cmsValidator = new CmsValidator(truststore, null);
+
+    // Because we don't have a CRL CertStore, revocation is not checked
+    assertThat(cmsValidator.validate(cms, data)).isTrue();
+  }
+
+  @Test
+  public void cmsValidationWithWrongSignedData() {
+    final CmsCreator cmsCreator = new CmsCreator(keystore, "trusted");
+    final Bytes otherData = Bytes.random(32);
```

Comment:
we don't really need to use random data to male sure that the data is different. Just use the same data and change one byte? 
otherData = Bytes.wrap(new byte[32]);
than set one byte to something other than 0 for the expectedData